File: app/src/main/java/com/katkov/lolachievements/application/app/LolApp.java
package com.katkov.lolachievements.application.app;

import android.app.Application;

import com.katkov.lolachievements.data.local.database.AppDataBase;
import com.katkov.lolachievements.di.Scopes;
import com.katkov.lolachievements.di.module.CiceroneModule;
import com.katkov.lolachievements.di.module.CommonModule;

import androidx.room.Room;
import toothpick.Scope;
import toothpick.Toothpick;

public class LolApp extends Application {

    private AppDataBase appDataBase;

    @Override
    public void onCreate() {
        super.onCreate();

        initDataBase();
        initDi();
    }

    private void initDi() {
        Scope appScope = Toothpick.openScope(Scopes.APP_SCOPE);
        appScope.installModules(new CommonModule(this));
        appScope.installModules(new CiceroneModule());

        Toothpick.inject(this, appScope);
    }

    private void initDataBase() {
        appDataBase = Room.databaseBuilder(this, AppDataBase.class, "database")
                .allowMainThreadQueries() //????
                .build();
    }
}

File: app/src/main/java/com/katkov/lolachievements/application/app/LolApp.kt
package com.katkov.lolachievements.application.app

import android.app.Application
import com.facebook.stetho.Stetho
import com.jakewharton.threetenabp.AndroidThreeTen
import com.katkov.lolachievements.di.Scopes
import com.katkov.lolachievements.di.module.AppDataBaseModule
import com.katkov.lolachievements.di.module.CommonModule
import com.katkov.lolachievements.di.module.GlobalCiceroneModule
import toothpick.Toothpick

class LolApp : Application() {

    override fun onCreate() {
        super.onCreate()

        initDi()
        initStetho()
        initThreeTenAbp()
    }

    private fun initDi() {
        Toothpick.inject(this, Toothpick.openScope(Scopes.APP_SCOPE).apply {
            installModules(CommonModule(this@LolApp))
            installModules(AppDataBaseModule(this@LolApp))
            installModules(GlobalCiceroneModule())
        })
    }

    private fun initStetho() = Stetho.initializeWithDefaults(this)

    private fun initThreeTenAbp() = AndroidThreeTen.init(this)
}
